gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/sql/GenericStatement.java

```diff
@@ -851,6 +852,20 @@ private final PreparedStatement prepMinion(
                                             }
                                           }
 
+                                          if (routeQuery && qinfo != null && qinfo.isDML() && qinfo.isInsert() && ((InsertQueryInfo)qinfo).isPutDML()) {
+                                            InsertNode in = qt instanceof InsertNode ? ((InsertNode)qt) : null;
+                                            TableDescriptor ttd = in != null ? in.targetTableDescriptor : null;
+                                            String table = ttd.getQualifiedName();
+                                            Region region = Misc.getRegionForTable(table.replaceAll("\"", ""), true);
+                                            GemFireContainer container = (GemFireContainer)region.getUserAttribute();
+                                            boolean isColumnTable = container.isRowBuffer();
+                                            if (isColumnTable) {
+                                              return getPreparedStatementForSnappy(false, statementContext, lcc,
+                                                  cc.isMarkedAsDDLForSnappyUse(), checkCancellation,
+                                                  DML_TABLE_PATTERN.matcher(source).find(), null);
+                                            }
+                                          }
```

Comment:
For call to getPreparedStatementForSnappy(), need to pass commitNestedTransaction as true. Also isUpdateOrDelete should be false for now. Can just just pass true instead of  DML_TABLE_PATTERN.matcher(source).find() once we add implementation for returning count. isDDL also should be passed just false instead of call to cc.isMarkedAsDDLForSnappyUse()

Reply:
Added implementation for returning count and. also incorporated above changes